### DISCUSSION TITLE
Memory requirements for running JBrowse2

### DISCUSSION THREAD

**Discussion by mickey-spongebob** [GMOD/jbrowse-components#3130] [Q&A] [answered]
Hi all,

Hope this finds you well! I managed to successfully setup a genome browser for our genome and it's gene and transcript annotation. The only problem I'm facing is that it is quite slow when searching and loading genes/transcripts. I'm using an Ubuntu 20.04.4 LTS machine with 16GB RAM and 8 CPUs. The genome size is ~1.5Gbp with 330 scaffolds and ~160,000 transcripts (including isoforms).

Is there a way I can speed up the search/lookup and loading function for the browser? 

I apologise if I missed a previous discussion. I looked for similar discussions but couldn't find one that matched.

Best,
kevin

**Comment by cmdcolin** [comment, marked as answer]
One thing that can happen is when all the transcript names are very similar e.g. PREFIX0000000123,PREFIX000000124 the indexing strategy groups them based on a set prefix size of 6 by default so all of them would end up in the same bucket and then it would search the whole bucket which is slow

You can manually tune this with --prefixSize in the jbrowse text-index command https://jbrowse.org/jb2/docs/cli/#jbrowse-text-index and in the case above it might want to be large like 11 or something

We may create an automatically tuned version of text-index so users don't have to manually configure this possibly. 

Let me know if that helps!

**Comment by mickey-spongebob** [comment]
Awesome!

I just followed your recommendation and it worked really well! Name searches are really fast now!

Thank you very much and best regards,
kevin
